openstax/middleware.py

```diff
@@ -39,3 +48,95 @@ def process_response(self, request, response):
             response = self.handler.get_response(request)
 
         return response
+
+
+class CommonMiddlewareOpenGraphRedirect(CommonMiddleware):
+    OG_USER_AGENTS = [
+        'twitterbot',
+        'facebookbot',
+        'pinterestbot',
+        'slackbot-linkexpanding',
+    ]
+
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request, *args, **kwargs):
+        if 'HTTP_USER_AGENT' in request.META:
+
+            user_agent = user_agent_parser.ParseUserAgent(request.META["HTTP_USER_AGENT"])
+            if user_agent['family'].lower() in self.OG_USER_AGENTS:
+                # url path minus the trailing /
+                url_path = request.get_full_path()[:-1]
+
+                full_url = request.build_absolute_uri()
+
+                # imdex of last / to find slug, except when there isn't a last /
+                if url_path == '':
+                    page_slug = "openstax-homepage"
+                else:
+                    index = url_path.rindex('/')
+                    page_slug = url_path[index+1:]
+
+                if self.redirect_path_found(url_path):
+                    # supporters page has the wrong slug
```

Comment:
it would probably be worth at least making a card to align these two slugs on BE/FE